src/github.com/getlantern/lantern-ui/app/js/app.js

```diff
@@ -122,15 +122,20 @@ var app = angular.module('app', [
           return;
         }
 
-        // try to reconnect indefinitely
-        // when the websocket closes
-        $interval(function() {
-          console.log("Trying to reconnect to disconnected websocket");
-          ds = $websocket('ws://' + document.location.host + '/data');
-          ds.onOpen(function(msg) {
-            $window.location.reload();
-          });
-        }, WS_RECONNECT_INTERVAL);
+        // Temporary workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1192773
+        if (navigator.userAgent.toLowerCase().indexOf('firefox') > -1) {
```

Comment:
Yes, that 40-50 secs. delay is the bug. It grows if you add more tabs, and there is a "memory" of opened tabs, so even if you close them it happens.
The only way I found the "Waiting for Lantern" message can be avoided when you run Lantern (if the bug has been triggered) is by preventing Firefox to reconnect automatically. So yes, the intention of this workaround is that once lantern is shut down and then restarted, then you need to open a new page or reload manually.